<launch>

  <arg name="classifierDir" default="Kinect"/>
  <arg name="replayRate" default="5.0"/>

  <!-- Launch the kinect for the data -->
  <!-- <include file="$(find kinect_data)/launch/kinect_data.launch"/>  -->

  <!-- Launch consisteny node because we need the ROIS-->
  <node pkg="consistency" type="consistency_node" name="consistency" >
	<param name="Q_Size" type="int" value="20"/>
	<!-- mode = {accumulate,load} -->
	<param name="UseDefaultRois" type="bool" value="true" />
	<param name="mode" type="string" value="load"/>
	<param name="label" 	 type="int"    value="1"  />
        <param name="yaml_filename" value="$(env NIST_CLASSIFIERS)/$(arg classifierDir)/constraints.yml"/>
	<param name="max_training_samples" type="int" value="2000"/>
	<remap from="input_rois" to="output_rois" />
	<remap from="Color_Image" to="/camera/rgb/image_color" />
	<remap from="Disparity_Image" to="/camera/depth_registered/disparity" />
        <remap from="PtCloud" to="/points" />
        <param name="Kinect_Disparity_Fix" value="true"/> 
   </node>


   <!-- Launch HaarAda Node-->
   <node name="HaarDispAdaNode" pkg="HaarDispAda" type="HaarDispAda174" output="screen">	
	<param name="Q_Size" type="int" value="20"/>
        <remap from="Color_Image" to="/ConsistencyColorImage"/>	
        <remap from="Disparity_Image" to="/ConsistencyDisparityImage"/>	
        <remap from="input_rois" to="ConsistencyOutputRois"/>
        <!-- mode={detect,train,load,accumulate} -->
        <param name="mode" value="load"/>
        <param name="classifier_file" value="$(env NIST_CLASSIFIERS)/$(arg classifierDir)/HaarDispAda.xml"/>
        <param name="num_Training_Samples" value="2000"/>
        <param name="HaarDispAdaPrior" value="7.5"/>
        <param name="Kinect_Disparity_Fix" value="false"/>
        <param name="UseMissingDataMask" value="false"/>
   </node> 

   <!-- display the rois -->
   <!-- <node pkg="roiViewer" type="roiViewer" name="roiViewer" >
        <param name="mode" type="string" value="roi_display"/> 
        <param name="label" type="int" value="1" />
	<remap from ="input_image" to="HaarDispAdaColorImage"/>
	<remap from ="input_rois" to="HaarDispAdaOutputRois"/>  
   </node> -->

</launch>